tests/e2e/high_availability_experimental/coord_cluster_registration.py

```diff
@@ -280,5 +284,129 @@ def check_coordinator2():
     mg_sleep_and_assert(expected_cluster_not_shared, check_coordinator2)
 
 
+# TODO: (andi) Test when dealing with distributed coordinators that you can register on one coordinator and unregister from any other coordinator
+@pytest.mark.parametrize(
+    "kill_instance",
+    [True, False],
+)
+def test_unregister_replicas(kill_instance):
+    safe_execute(shutil.rmtree, TEMP_DIR)
```

Comment:
Can you add steps here what we do in each step, it is easier to get intention just from reading description